src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/LicencePolicyController.kt

```diff
@@ -142,20 +142,20 @@ class LicencePolicyController(
   fun compareLicence(
     @PathVariable("version") version: String,
     @PathVariable("licenceId") licenceId: Long,
-  ): List<LicenceConditionChanges>? {
-    val currentLicence = licenceService.getLicenceById(licenceId)
-    val currentPolicy = licencePolicyService.policyByVersion(version)
+  ): List<LicenceConditionChanges>? = when (val currentLicence = licenceService.getLicenceById(licenceId)) {
+    is ModelVariation -> getPolicyChanges(currentLicence, currentLicence.variationOf, version)
 
-    return when {
-      currentLicence !is VariationLicence -> emptyList()
-      else -> getPreviousPolicy(currentLicence)?.let {
-        licencePolicyService.compareLicenceWithPolicy(currentLicence, it, currentPolicy)
-      } ?: emptyList()
-    }
+    else -> emptyList()
   }
 
-  private fun getPreviousPolicy(currentLicence: VariationLicence): LicencePolicy? {
-    val previousLicence = currentLicence.variationOf?.let { licenceService.getLicenceById(it) }?.version
-    return previousLicence?.let { licencePolicyService.policyByVersion(it) }
+  fun getPolicyChanges(currentLicence: ModelVariation, parentLicenceId: Long?, version: String): List<LicenceConditionChanges> {
+    if (parentLicenceId === null) return emptyList()
+    val parentLicence = licenceService.getLicenceById(parentLicenceId)
+
+    return licencePolicyService.compareLicenceWithPolicy(
+      currentLicence,
+      parentLicence,
+      version,
+    )
```

Comment:
(this is all a bit messy which is why its difficult to work with - ideally we'd sort it out properly and move out into a service, fix the potential circular dependency and  work on the licence entity rather than the model object but all for another time!)